src/Config/RuntimeConfigLoader.cs

```diff
@@ -24,6 +24,8 @@ public abstract class RuntimeConfigLoader
 {
     protected readonly string? _connectionString;
 
+    public RuntimeConfig? RuntimeConfig;
```

Comment:
can you add comment here and details in pr description describing the `RuntimeConfigLoader` API changes: 
1. Why is `RuntimeConfig` public, could it be protected?
2. Could the function `TryParseConfig` which has `out` paramter supplying runtime config also save state?

Reply:
added a comment. This is something we should consider for a future refactor. But for the moment would like to avoid this can of worms if possible.